### PR TITLE
[SYCL free func] Refactor WelfordNorm by free function

### PR DESCRIPTION
**Pull request by minmingzhu**
# WelfordNorm.h — Kernel UT Report

- SYCL kernel conversion check: In `src/ATen/native/xpu/sycl/WelfordNorm.h`, `welford_batch_norm_stat_channels_last_vec_kernel` is implemented as a free function via `SYCL_EXT_ONEAPI_FUNCTION_PROPERTY((sycl::ext::oneapi::experimental::nd_range_kernel<2>))` (lines 106–108), and is submitted through the kernel function pointer in `WelfordBatchNormStatChannelsLastVecKernelConfig` (lines 254–258).

- There are no remaining direct SYCL launch functors for this kernel. `WelfordBatchNormStatChannelsLastVecKernelConfig` is launch-configuration state, not a kernel functor.

- Changed-kernel related UTs run in the current PyTorch/XPU integration:

1. In PyTorch (`/workspace1/minminz/workpath/pytorch-public`):
   - Command: `pytest /workspace1/minminz/workpath/torch-xpu-ops/test/xpu/test_nn_xpu.py -v -k "batch_norm or batchnorm or BatchNorm"`
   - Result: **94 passed, 17 skipped, 3392 deselected**.
2. XPU build:
   - Command: `pip install -e . -v --no-build-isolation`
   - Result: completed successfully.

### CODE REVIEW

**Comment by tye1** [comment]
@torchxpubot  check-ut

File: src/ATen/native/xpu/sycl/WelfordNorm.h [hunk 1]
```diff
@@ -101,161 +102,161 @@ template <
...
-  void operator()(sycl::nd_item<2> item) const {
-    //  init private counter
-    acc_vec_t mean;
-    acc_vec_t m2n;
-    int_vec_t count;
+  auto item = sycl::ext::oneapi::this_work_item::get_nd_item<2>();
+  auto local_size = item.get_local_range(0) * item.get_local_range(1);
+
+  char* scratch =
+      (char*)sycl::ext::oneapi::experimental::get_work_group_scratch_memory();
```
Comment (gujinghui) [inline src/ATen/native/xpu/sycl/WelfordNorm.h]:
```suggestion
      static_cast<char*>(sycl::ext::oneapi::experimental::get_work_group_scratch_memory());
```

**Comment by minmingzhu** [comment]
@torchxpubot check-ut

**Comment by tye1** [comment]
@gujinghui  any further comments? thanks.

**Comment by minmingzhu** [comment]
@torchxpubot check-ut

**Comment by tye1** [comment]
@torchxpubot  check-ut

**Review by gujinghui — APPROVED** [review]
